Nix build for a 'smart' docker image that can load keyfiles or a pier and boot a ship (#4256)
urbit/urbit: pkg/docker-image/README.md

# Official Urbit Docker Image

This is the official Docker image for [Urbit](https://urbit.org). 

Urbit is a clean-slate OS and network for the 21st century.

## Using

To use this image, you should mount a volume with a keyfile, comet file, or existing pier at `/urbit`, and map ports
as described below.

### Volume Mount

> This image expects a volume mounted at `/urbit`. This volume should initially obtain one of

Reviewer: ```suggestion
This image expects a volume mounted at `/urbit`. This volume should initially contain one of
```

- A keyfile `<shipname>.key` for a galaxy, star, planet, or moon. See the setup instructions for Urbit for information on [obtaining a keyfile](https://urbit.org/using/install/). 
  * e.g. `sampel-palnet.key` for the planet `sampel-palnet`.

>   to urbit to boot a ship from an existing pier, and in most cases (other than comets) will be named for the ship.

Reviewer: I'm having trouble understanding what exactly is meant by this sentence—could this be rephrased?

Author: I think this was a spurious paste as a result of writing docs too late at night. I've removed it.

- An empty file with the extension `.comet`. This will cause Urbit to boot a [comet](https://urbit.org/docs/glossary/comet/) in a pier named for the `.comet` file (less the extension).
  * e.g. starting with an empty file `my-urbit-bot.comet` will result in Urbit booting a comet into the pier
    `my-urbit-bot` under your volume.
- An existing pier as a directory `<shipname>`. You can migrate an existing ship to a new docker container in this way by placing its pier under the volume.

>   * e.g. if your ship is `sampel-palnet` then you likely have a directory `sampel-palnet` whose path you pass to `./urbit` when starting. While your ship is not running, move this directory to the volume and then start the container.

Reviewer: ```suggestion
  * e.g. if your ship is `sampel-palnet` then you likely have a directory `sampel-palnet` whose path you pass to `./urbit` when starting. [Move your pier](https://urbit.org/using/operations/using-your-ship/#moving-your-pier) directory to the volume and then start the container.
```
